contribs/src/main/java/com/netflix/conductor/contribs/listener/archive/ArchivingWorkflowListenerProperties.java

```diff
@@ -12,46 +12,51 @@
  */
 package com.netflix.conductor.contribs.listener.archive;
 
-import org.springframework.beans.factory.annotation.Value;
-import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.ConfigurationProperties;
 import org.springframework.core.env.Environment;
-import org.springframework.stereotype.Component;
 
-@Component
-@ConditionalOnProperty(prefix = "workflow", name = "status.listener.type", havingValue = "archive")
+@ConfigurationProperties("conductor.workflow-status-listener.archival")
 public class ArchivingWorkflowListenerProperties {
 
     private final Environment environment;
 
+    @Autowired
     public ArchivingWorkflowListenerProperties(Environment environment) {
         this.environment = environment;
     }
 
     /**
      * The time to live in seconds for workflow archiving module. Currently, only RedisExecutionDAO supports this
      */
-    @Value("${workflow.archival.ttl.seconds:0}")
-    private int workflowArchivalTTL;
+    private int ttlSeconds = 0;
```

Comment:
The property could be of type Duration. Not required, but maybe worth considering, https://docs.spring.io/spring-boot/docs/current/reference/html/spring-boot-features.html#boot-features-external-config-conversion-duration

Reply:
Good idea, I will do this in a separate PR, since this one is already so huge. It will also make this easier to document.